packages/components/src/components/post-megadropdown/post-megadropdown.tsx

```diff
@@ -46,18 +46,58 @@ export class PostMegadropdown {
    **/
   @Event() postToggleMegadropdown: EventEmitter<{ isVisible: boolean; focusParent?: boolean }>;
 
+  private defaultSlotObserver: MutationObserver;
+
+  private checkInitialAriaCurrent() {
+    const currentEl = this.host.querySelector('[aria-current="page"]');
+    if (!currentEl) return;
+
+    const hostId = this.host.getAttribute('id');
+    if (!hostId) return;
+
+    const triggerButton = document.querySelector(
+      `post-megadropdown-trigger[for="${hostId}"] > button`,
+    );
+
+    if (triggerButton) {
+      triggerButton.classList.add('selected');
+    }
+  }
+
+  private setupObserver() {
+    if (this.defaultSlotObserver) return;
+
+    const config: MutationObserverInit = {
+      subtree: true,
+      attributes: true,
+      attributeFilter: ['aria-current'],
+    };
+
+    this.defaultSlotObserver = new MutationObserver(this.handleAriaCurrentChange.bind(this));
+    this.defaultSlotObserver.observe(this.host, config);
+  }
+
   disconnectedCallback() {
     this.removeListeners();
     window.removeEventListener('postBreakpoint:device', this.breakpointChange.bind(this));
     if (PostMegadropdown.activeDropdown === this) {
       PostMegadropdown.activeDropdown = null;
     }
+    if (this.defaultSlotObserver) {
+      this.defaultSlotObserver.disconnect();
+    }
```

Comment:
You created it so you know it exists: 
```suggestion
     this.defaultSlotObserver.disconnect();
```